### PR TITLE
Edits where the Preview Mode Sign In Prompt buttons send you to

### PR DESCRIPTION
**Pull request by Kokobe**
Makes the onboarding buttons in the Preview Mode Sign In Prompt send you to the WelcomeFlowActivity instead of the old FirstRunActivity. 

### Pointers to help with review
* https://neevaco.slack.com/archives/C02RYG8LY8H/p1677573443254419

### How was this tested?
- [x] I tested this manually on `debug`
- [ ] I tested this manually on `staging` or `release`
- [ ] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes
- [ ] I checked that `@Previews` still work in Android Studio

### Manual test cases
Forced the preview mode sign in prompt to open on init. Saw that the buttons opened WelcomeFlowActivity.

### Screenshots and screen recordings

https://user-images.githubusercontent.com/20916043/222253378-86cecb78-619b-46f6-9e1a-2c8089e8219f.mp4

### CODE REVIEW

File: app/src/main/java/com/neeva/app/firstrun/signup/SignUpLandingScreen.kt [hunk 1]
```diff
@@ -45,6 +47,7 @@ fun SignUpLandingContainer(
     ) { modifier ->
         SignUpLandingScreen(
             onOpenUrl = onOpenUrl,
+            showSignUpWithGoogle = {},
```
Comment (Kokobe) [inline app/src/main/java/com/neeva/app/firstrun/signup/SignUpLandingScreen.kt:50]: This composable is only used in FirstRunActivity which I will delete in a follow-up PR

File: app/src/main/java/com/neeva/app/firstrun/signup/PreviewModeSignUpPrompt.kt [hunk 2]
```diff
@@ -42,8 +42,18 @@ fun PreviewModeSignUpPrompt(
...
+                    screenToReturnTo = AppNavDestination.BROWSER,
+                    purpose = WelcomeFlowNavModel.Companion.Purpose.SIGN_UP
+                )
+                onDismiss()
+            },
             showSignUpWithOther = {
-                signInFlowNavModel.navigateToSignUpWithOther()
+                appNavModel.showWelcomeFlow(
+                    screenToReturnTo = AppNavDestination.BROWSER,
+                    purpose = WelcomeFlowNavModel.Companion.Purpose.SIGN_IN
```
Comment (dan-alcantara-neeva) [inline app/src/main/java/com/neeva/app/firstrun/signup/PreviewModeSignUpPrompt.kt:55]: Why is this "SIGN_IN" if you're "signing up with other?

Comment (Kokobe) [reply app/src/main/java/com/neeva/app/firstrun/signup/PreviewModeSignUpPrompt.kt:55]: Ah it should've been `SIGN_UP`. Both lead to the Plans screen first if Billing is enabled, so it's okay to send them there.

**Comment by Kokobe** [comment]
Updated the video too

**Review by dan-alcantara-neeva — APPROVED** [review]

**Pull request merged**